### PR TITLE
Dashboards

### PR DESCRIPTION
**Pull request by VassilIordanov**
closes #1494 

### User changes
- Users have access to dashboard from the navigation menu

### System admin changes
- Sys admins can add new dashboards by modifying anet.yml
- Sys admins can modify existing dashboards, by editing the data files

- [x] anet.yml needs change
- [x] documentation has changed

### CODE REVIEW

File: src/main/java/mil/dds/anet/beans/Task.java [hunk 1]
```diff
@@ -178,6 +184,16 @@ public Organization getResponsibleOrg() {
...
 	}
 
+	@GraphQLQuery(name="reports")
+	public CompletableFuture<AnetBeanList<Report>> loadReports(@GraphQLRootContext Map<String, Object> context, @GraphQLArgument(name="query") ReportSearchQuery query) {
+	// TODO: Use the query parameter 
+		if (reports != null) {
+			return CompletableFuture.completedFuture(reports);
+		}
+		return AnetObjectEngine.getInstance().getTaskDao().getReportsForTask(context, uuid)
+				.thenApply(o -> { reports = new AnetBeanList<Report>(o); return reports; });
```
Comment (VassilIordanov) [inline src/main/java/mil/dds/anet/beans/Task.java]: When only a totalCount is needed, there is probably a better way to get this from the ResultSet. Could be good to make `AnetBeanList.getList` lazy, and be the one that triggers parsing of the Report beans

File: src/main/java/mil/dds/anet/AnetApplication.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public DataSourceFactory getDataSourceFactory(AnetConfiguration configuration) {
 
     // Serve assets on /assets
     bootstrap.addBundle(new ConfiguredAssetsBundle(ImmutableMap.<String, String>builder()
-        .put("/assets/", "/assets/").put("/imagery/", "/imagery/").build(), "index.html"));
+        .put("/assets/", "/assets/").put("/imagery/", "/imagery/").put("/data/", "/data/").build(), "index.html"));
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/AnetApplication.java]: Code line is improperly formatted.

File: client/src/pages/dashboards/DecisivesDashboard.js [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+                )}
+              />
+            ))}
+          </Panel.Body>
+        </Panel>
+        <Panel>
+          <Panel.Heading>
+            <Panel.Title componentClass="h3">Processes</Panel.Title>
+          </Panel.Heading>
+          <Panel.Body>{/* <StatsTable data={[]} /> */}</Panel.Body>
```
Comment (gjvoosten) [inline client/src/pages/dashboards/DecisivesDashboard.js]: This part of the dashboard would appear to be always empty.

File: anet.yml [hunk 4]
```diff
@@ -400,6 +407,17 @@ dictionary:
           layers: GEBCO_LATEST
           format: "image/png"
 
-  automaticallyInactivateUsers:
-    emailRemindersDaysPrior: [15, 30, 45]
-    ignoredDomainNames: []
+#  automaticallyInactivateUsers:
+#    emailRemindersDaysPrior: [15, 30, 45]
+#    ignoredDomainNames: []
```
[the remark this answers is not in the document]
Comment (VassilIordanov) [reply anet.yml]: actually yes, there was a misunderstanding of what `domain` means in the requirement of this implementation, so it is not currently in use

File: client/src/pages/dashboards/DecisivesDashboard.js [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+          .subtract(1, "d")
+          .endOf("day")
+          .valueOf()
+      },
+      this.deserializeCallback
+    )
+    this.fetchStaticData()
+  }
+
+  @autobind
```
[the remark this answers is not in the document]
Comment (VassilIordanov) [reply client/src/pages/dashboards/DecisivesDashboard.js]: thanks

File: client/src/components/graphs/Pie.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
+import PropTypes from "prop-types"
+import * as d3 from "d3"
+import SVGCanvas from "components/graphs/SVGCanvas"
```
[the remark this answers is not in the document]
Comment (VassilIordanov) [reply client/src/components/graphs/Pie.js]: +1

File: client/src/pages/dashboards/KanbanDashboard.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import Page, {
+  mapDispatchToProps,
+  propTypes as pagePropTypes
+} from "components/Page"
+import GQL from "graphqlapi"
+import { Task } from "models"
+import React from "react"
+import Kanban from "components/Kanban"
+import { Settings } from "api"
```
[the remark this answers is not in the document]
Comment (VassilIordanov) [reply client/src/pages/dashboards/KanbanDashboard.js]: :+1:

File: client/src/pages/tasks/Show.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import { Settings } from "api"
 import AppContext from "components/AppContext"
 import * as FieldHelper from "components/FieldHelper"
+// import Calendar from 'components/Calendar'
```
[the remark this answers is not in the document]
Comment (VassilIordanov) [reply client/src/pages/tasks/Show.js]: oops, this should not have been in there

File: src/main/resources/migrations.xml [hunk 9]
```diff
@@ -2279,6 +2279,12 @@
 		<!-- Rolling back is not useful -->
 	</changeSet>
 
+	<changeSet id="add-assessment-to-Tasks" author="VassilIordanov">
```
[the remark this answers is not in the document]
Comment (VassilIordanov) [reply src/main/resources/migrations.xml]: I am afraid this is the correct position vis-a-vis production ;-)

File: src/main/java/mil/dds/anet/AnetApplication.java (hunk 2 above)
Comment (VassilIordanov) [reply src/main/java/mil/dds/anet/AnetApplication.java]: :+1:

File: client/src/pages/dashboards/DecisivesDashboard.js (hunk 3 above)
Comment (VassilIordanov) [reply client/src/pages/dashboards/DecisivesDashboard.js]: yeah, waiting for more requirements on that one... still work in progress

File: client/src/pages/tasks/Show.js [hunk 10]
```diff
@@ -125,16 +129,18 @@ class BaseTaskShow extends Page {
                   title={`${Settings.fields.task.shortLabel} ${task.shortName}`}
                   action={action}
                 />
+                {/* <Calendar> </Calendar> */}
```
[the remark this answers is not in the document]
Comment (VassilIordanov) [reply client/src/pages/tasks/Show.js]: :+1:

**Pull request merged**